homeassistant/components/tplink_omada/switch.py

```diff
@@ -108,8 +112,7 @@ async def async_turn_off(self, **kwargs: Any) -> None:
 
     def _refresh_state(self) -> None:
         self._attr_is_on = self.port_details.poe_mode != PoEMode.DISABLED
-        if self.hass:
-            self.async_write_ha_state()
+        self.async_write_ha_state()
```

Comment:
It's better to remove the state update here and put it in `_handle_coordinator_update` instead, eg by calling super.

There's already a state update at the end of entity addition.

Reply:
Thanks for the feedback. I'm still learning a lot of the framework, so corrections are much appreciated. I can look at improving this in the next version, as there is quite a lot of refactoring to do to add other domains for the network switches features.